barretenberg/cpp/src/barretenberg/proof_system/op_queue/ecc_op_queue.hpp

```diff
@@ -302,6 +295,74 @@ class ECCOpQueue {
         ultra_ops[2].emplace_back(tuple.z_1);
         ultra_ops[3].emplace_back(tuple.z_2);
     }
+
+  private:
+    /**
+     * @brief TESTING PURPOSES ONLY: Populate ECC op queue with mock data as stand in for "previous circuit" in tests
+     * @details TODO(#723): We currently cannot support Goblin proofs (specifically, transcript aggregation) if there
+     * is not existing data in the ECC op queue (since this leads to zero-commitment issues). This method populates the
+     * op queue with mock data so that the prover of an arbitrary 'first' circuit can behave as if it were not the
+     * prover over the first circuit in the stack. This method should be removed entirely once this is resolved.
+     *
+     * @param op_queue
+     */
+    void populate_with_mock_initital_data()
```

Comment:
So I change the mock function to do the valid thing without using a circuit builder, it's a bit complicated but I like having this isolated as low as possible and only having one mocking function. Wdyt?